src/VASSAL/counters/Deck.java

```diff
@@ -1361,12 +1329,10 @@ private File getSaveFileName() {
     File sf = fc.getSelectedFile();
     if (sf != null) {
       String name = sf.getPath();
-      if (name != null) {
```

Comment:
Is `File.getPath()` guaranteed to return nonnull?

Reply:
Yes it is guaranteed, a `File`'s path can only be set when constructing and `File` has null checks in all it's constructors and will throw an NPE if it encounters a null path.